src/main/java/io/pravega/connectors/hadoop/PravegaInputFormat.java

```diff
@@ -77,15 +92,24 @@ protected PravegaInputFormat(ClientFactory externalClientFactory) {
         final String deserializerClassName = Optional.ofNullable(conf.get(PravegaInputFormat.DESERIALIZER)).orElseThrow(() ->
                 new IOException("The event deserializer must be configured (" + PravegaInputFormat.DESERIALIZER + ")"));
 
+        final StreamCut startStreamCut = getStreamCutFromPositionsJson(
+                scopeName,
+                streamName,
+                Optional.ofNullable(conf.get(PravegaInputFormat.START_POSITIONS)).orElse(""));
+        final StreamCut endStreamCut = getStreamCutFromPositionsJson(
+                scopeName,
+                streamName,
+                Optional.ofNullable(conf.get(PravegaInputFormat.END_POSITIONS)).orElse(""));
+
         ClientFactory clientFactory = (externalClientFactory != null) ? externalClientFactory : ClientFactory.withScope(scopeName, controllerURI);
 
         // generate a split per segment
         List<InputSplit> splits = new ArrayList<InputSplit>();
         BatchClient batchClient = clientFactory.createBatchClient();
-        for (Iterator<SegmentInfo> iter = batchClient.listSegments(new StreamImpl(scopeName, streamName)); iter.hasNext(); ) {
-            SegmentInfo segInfo = iter.next();
-            Segment segment = segInfo.getSegment();
-            PravegaInputSplit split = new PravegaInputSplit(segment, segInfo.getStartingOffset(), segInfo.getWriteOffset());
+        Iterator<SegmentRange> iter = batchClient.getSegments(new StreamImpl(scopeName, streamName), startStreamCut, endStreamCut).getIterator();
```

Comment:
we can use `Stream.of(scopeName, streamName)` instead.

Reply:
thanks, accepted